data/src/buffer.rs

```diff
@@ -357,6 +359,41 @@ impl From<SkinTone> for emojis::SkinTone {
     }
 }
 
+#[derive(Clone, Copy)]
+pub struct Previews<'a> {
+    collection: &'a preview::Collection,
+    cards_are_visible: bool,
+    images_are_visible: bool,
+}
+
+impl<'a> Previews<'a> {
+    pub fn from(
+        collection: &'a preview::Collection,
+        target: &Target,
+        config: &config::Preview,
+        casemapping: isupport::CaseMap,
+    ) -> Previews<'a> {
```

Comment:
Nit: I wouldn't use from as name since it collides with From implementation. Maybe just `new`